README.md

```diff
@@ -9,3 +9,23 @@ Primary Goals:
   - src/main/java, src/main/test
   - relatively coarse-grained modules
 - testing: various real-ish scenarios are present in this project, and usable.
+
+
+Bazel Target Naming Conventions
+
+A lot fot hese are repeated across modules, and bazel module scoping
+provides enough namespacing information that it's redundant and arguably
+confusing to repeat the namespace in such target names. This system
+also has the benefits of being consistent horizontally and therefore
+is easily grep-able and so on.
+
+- main code for module: "lib"
```

Comment:
should this be java_lib ?

Reply:
I dunno. `java_lib`, as opposed to what?

also would you convert `test_lib` to be `test_java_lib`?